### PR TITLE
[Hub Generated] Review request for Microsoft.Maps to add version stable/2021-02-01

### PR DESCRIPTION
**Pull request by stack111**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/stack111/azure-rest-api-specs/dev-maps-Microsoft.Maps-2020-02-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.

**April 2021**

  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.

**April 2021**

  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [x] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jorgecotillo** [comment]
Just making sure that this PR is removing endpoints, is it expected?

**Comment by stack111** [comment]
> his PR is removing endpoints, is it expected?

Yes, removing privateAtlas is expected, we are retiring the resourceType and it being replaced with "creators" type.

**Comment by stack111** [comment]
@jorgecotillo is there some action which I need to take here? This PR is addressing a bunch of PLR criteria which is on deadline, I see dotnet sdk build is failing but I'm not sure if that's caused from my PR.

**Comment by stack111** [comment]
@ArcturusZhang - is there any action which is required from the dotnet sdk build failure here for me?

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:531]: you should still support a 200 in case of no-op PUTs for example, or PUTs to update tags or some such that can be done snchronously

Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:531]: do you mean Creating?

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:625]: so there is no scenario in which a delete can be synchronously completed? adding this later will require a new api-version

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:754]: same as above, can remove

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:444]: Correct, we have 90 day notice plan in place to deprecate in in August the preview api. Microsoft.maps/accounts/creators is the replacement resource for privateAtlases.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1068]: >x-ms-client-id [](http://example.com/codeflow?start=9&length=14)

this is the naming used for headers, not model properties

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1097]: you can specify min, max for integer values I believe

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1078]: specify the valid values via enum and x-ms-enum

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateAccount.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateAccount.json:45]: provide realistic values

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: this is not a valid arm region, please use eastus or similar

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (zhugexubin) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: What is a no-op PUTs? I thought update tags is through patch API instead of PUT.

Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: Yes I think we do, I actually updated the description to provide the target state only so there is confusion.

Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: I updated and forwarded this to the team.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:546]: I updated and forwarded this to the team.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/ListKeys.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/ListKeys.json:11]: not sure why you need this in the response - since the request url contains it too

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:637]: correct, that is the behavior.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: fixed

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (zhugexubin) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:546]: I am following the RPC here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: fixed

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:245]: even if you think you don't need it now - its always good to support paging in the schema even if at runtime you don't

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (zhugexubin) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:637]: I think we will only support the async creator delete.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (varunkumta) [inline specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:828]: as mentioned above too - adding a nextLink here is good practice even if you don't need it right now

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1078]: This is considered a breaking change though, does it have to be done now?

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: Creator resource doesn't support valid ARM regions. It supports "unitedstates" and "europe". I understand Azure resource should support valid ARM regions however creator is considered "geographic" and not "regional" so a decision was made to support non-standard ARM regions in location. Let follow up on this.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1088]: We want our resource type to remain flexible here though. Initially, we only will be adding provisioningState for future proofing and return `Succeeded`. So we haven't yet designed what the possible non-terminal states will be and we know that in general other RP definitions are using only string and [RPaaS validates](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r4031-rpaasresourceprovisioningstate) for only `string`.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:1078]: We're considering uniqueId or uniqueIdentifier.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: Is there any issue with this at the moment? We will likely need an exception, until the team can move to standard ARM regions.

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: added name, but the value from the RP will be null

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json:824]: added name, but the value from the RP will be null

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/ListKeys.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/ListKeys.json:11]: Good point, I also noticed our schema is missing 2 properties. Will remove id and add missing properties.

Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/ListKeys.json]: Fixed and added missing properties.

**Comment by stack111** [comment]
@varunkumta as per your comments made changes, but now we have cross version breaking changes. Any recommendations what I can do here?

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json
Comment (varunkumta) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: I don't think routing will work in ARM for regions that ARM does not support. Since 'location' has special meaning at the top-level in a resource you are restricted to what ARM supports. The alternative would be to model this as a proxy resource with a customLocation property within the 'properties' section

Comment (varunkumta) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: see [https://armwiki.azurewebsites.net/rp_onboarding/tracked_vs_proxy_resources.html?q=tracked](https://armwiki.azurewebsites.net/rp_onboarding/tracked_vs_proxy_resources.html?q=tracked)

Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/examples/CreateMapsCreator.json:9]: See our existing prod manifest for [Microsoft.Maps/accounts/privateAtlases](https://armmanifest.visualstudio.com/One/_git/Manifest?path=%2FMICROSOFT.MAPS%2FPROD%2FMANIFEST.JSON&version=GBmaster&line=227&lineEnd=228&lineStartColumn=13&lineEndColumn=15&lineStyle=plain&_a=contents) and [Microsoft.Maps/accounts](https://armmanifest.visualstudio.com/One/_git/Manifest?path=%2FMICROSOFT.MAPS%2FPROD%2FMANIFEST.JSON&version=GBmaster&line=66&lineEnd=67&lineStartColumn=13&lineEndColumn=15&lineStyle=plain&_a=contents). 

Note 
- We have routing working however it results in single wildcard `""` and `"United States"` as location, even our top level resource is has historically been `"Global"` and we are progressing to ARM regions. 
- Creators type will follow the same pattern but later. Switching to proxy type doesn't address the problem because we want to leverage the support of a tracked resource. 
- in other comments, I mentioned "Creators" resource type is **replacing** the deprecated "privateAtlases" type.

As referenced, manifest routing by "united states" **works** but there are known limitations / constraints at the moment including managed identity (`identity` property) which relies on standard ARM regions.

Is this a blocker?

File: specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json
Comment (stack111) [reply specification/maps/resource-manager/Microsoft.Maps/stable/2021-02-01/maps-management.json]: I'm re-opening this @varunkumta, we found a case where we need to implement operationResults resource type for this to function e2e. In-light of that discovery, we're making a change here to only support the synchronous status codes + provisioningState described on the RPC addendum.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by stack111** [comment]
@ArcturusZhang 

Business Justification:

MAPS Creator is a PLR item that was recently onboarded. The GA is 5/21, but will use a new swagger API version containing the new SKUs. Once that swagger is merged we can kick off the GA work for Azure CLI, PowerShell and Terraform.

Creator is a resource moving from Preview to GA. It has not been part of a SDK release previously, so no customers SDK customers should be broken as a result of the change. The change removes an incorrect pattern and moves the RP to the ARM recommended best practice in preparation for the GA release. Can you please look and sign off?

**Comment by msyyc** [comment]
For python, the breaking change is ok

**Comment by ArcturusZhang** [comment]
Hi @stack111 I understand, but unfortunately I cannot sign off on the breaking changes issue - it can only be signed off by the breaking change review board. Could you please follow the guidance above to open a breaking change review request with your justification?

**Review by JeffreyRichter — APPROVED** [review]

**Comment by stack111** [comment]
@ArcturusZhang - we have the breaking changes approved. Please continue forward.

**Comment by stack111** [comment]
@ArcturusZhang - sorry I accidently clicked the wrong button and closed the PR, I reopened but it added the CI python tag back. Please approve as soon as we can.

File: package-lock.json
Comment (ArcturusZhang) [inline package-lock.json:1]: Could we revert the change of this file? We should not change this in a swagger PR.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Comment by ArcturusZhang** [comment]
We could merge this once we revert the changes on `package.lock` @stach111

File: package-lock.json
Comment (stack111) [reply package-lock.json]: Done

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
